api/envoy/config/cluster/v4alpha/cluster.proto

```diff
@@ -318,6 +318,23 @@ message Cluster {
     // The number of random healthy hosts from which the host with the fewest active requests will
     // be chosen. Defaults to 2 so that we perform two-choice selection if the field is not set.
     google.protobuf.UInt32Value choice_count = 1 [(validate.rules).uint32 = {gte: 2}];
+
+    // The following formula is used to calculate the dynamic weights when hosts have different
+    // load balancing weights:
+    //
+    // `weight = load_balancing_weight / (active_requests + 1)^active_requests_exponent`
+    //
+    // When `active_requests_exponent == 0.0` the Least Request Load Balancer doesn't consider the
+    // number of active requests at the time it picks a host and behaves like the Round Robin Load
+    // Balancer.
+    //
+    // When `active_requests_exponent > 0.0` the Least Request Load Balancer scales the load
+    // balancing weight by the number of active requests at the time it does a pick.
+    //
+    // The exponent is cached for performance reasons and its value is only refreshed whenever
+    // one of the Load Balancer's host sets changes, e.g., whenever there is a host membership
+    // update or a host load balancing weight change.
+    core.v4alpha.RuntimeDouble active_requests_exponent = 2;
```

Comment:
API change looks good, but one question: do you think the formula might change over time? If so, would it make sense to have this function be some sort of extension point, or even something like a CEL expression?

I don't want to over-design or build something that won't be needed here, the main concern I'm pointing out is that a formula like this might need to be tweaked, and stable APIs aren't the most exciting place to do this.

Reply:
That's a good point, but to mitigate the risk of API being tied super closely with a formula, we should just call the knob what it actually is. The exponent just gives us a knob to make the active requests have more/less impact on the weights, so I think we should just call it something along the lines of what it actually does and remove the word "exponent" in case the future holds a better formula to pull this off.

Reply:
@htuch / @tonya11en: the formula hasn't changed in two years, so I don't expect it to change often. In any case I think that `active_requests` will most probably always be a variable.

I suggest renaming the field to `active_requests_bias` which would be acceptable even if the formula changes to something like: `dynamic_weight = host_weight /  (1 + active_requests_bias * active_requests)`. An alternative is `active_requests_weight`, but it might be confusing.

Reply:
Ack, if it's going to be relatively stable then this is OK. Another naming suggestion would be `active_requests_factor`, although it's not really a factor when used as exponent. Feel free to leave as is or change if you have a reasonable replacement.